lib/invoker/power/setup.rb

```diff
@@ -1,15 +1,25 @@
 require "eventmachine"
+require "invoker/power/tld_validator"
 
 module Invoker
   module Power
     class Setup
       attr_accessor :port_finder
-      def self.install
+
+      def self.install(options = {})
+        if options[:tld]
+          validate_tld(options[:tld])
+          Invoker::Power::Setup.tld = options[:tld]
+        end
+
```

Comment:
this is how I would do:

```
`selected_installedr_class = installer_klass
installer = selected_installer_class.new(tld)
installer.install
```

So as `tld` itself becomes an attribute of installer class.